client/client-lib/src/lib.rs

```diff
@@ -577,6 +577,43 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
         Ok(final_notes)
     }
 
+    /// Select notes of a given amount, and then remint them
+    ///
+    /// This is a respin of `spent_ecash`, and I can't make sense
+    /// of the purpose. IMO, the whole client API needs deep
+    /// rethinking. --dpc
+    pub async fn remint_ecash<R: RngCore + CryptoRng>(&self, amount: Amount, rng: R) -> Result<()> {
+        let notes = self.mint_client().select_notes(amount).await?;
+
+        let mut tx = TransactionBuilder::default();
+
+        let (mut keys, input) = MintClient::ecash_input(notes)?;
+        tx.input(&mut keys, input);
+        self.submit_tx_with_change(tx, rng).await?;
+
+        Ok(())
+    }
+
+    /// Continuation of `remint_notes`
+    pub async fn remint_ecash_await(&self, amount: Amount) -> Result<TieredMulti<SpendableNote>> {
+        self.fetch_all_notes().await;
+        let notes = self.mint_client().select_notes(amount).await?;
+        assert_eq!(notes.total_amount(), amount, "should have exact change");
```

Comment:
1. `should have exact change` isn't super correct, it's not change but "input notes" we want to have in the right denominations
2. Compared to `spend_ecash` it seems accidental that we get the right notes (mostly because we ensure to always have at least n of each denomination after @jkitman's change)

But it seems to work and a new client will be able to model this way better. I would advise against using this in production though. Maybe add some comment that it's tests-only?

Reply:
I thought I copied `spend_ecash` 1:1, just split it into two parts.